src/Cli/dotnet/commands/dotnet-workload/list/WorkloadListCommand.cs

```diff
@@ -79,39 +108,36 @@ public override int Execute()
             return 0;
         }
 
-        private UpdateAvailableEntry[] MockUpdateAvailable()
+        internal UpdateAvailableEntry[] GetUpdateAvailable(IEnumerable<WorkloadId> installedList)
         {
-            var updateList = new List<UpdateAvailableEntry>();
-
-            if (!File.Exists(Path.Combine(MockUpdateDirectory,
-                "Microsoft.NET.Workload.Android.6.0.100.nupkg")))
-            {
-                updateList.Add(new UpdateAvailableEntry("6.0.100", "6.0.101",
-                    _mockAndroidDescription,
-                    "microsoft-android-sdk-full"));
-            }
-
-            if (!File.Exists(Path.Combine(MockUpdateDirectory,
-                "Microsoft.iOS.Bundle.6.0.100.nupkg")))
+            HashSet<WorkloadId> installedWorkloads = installedList.ToHashSet();
+            _workloadManifestUpdater.UpdateAdvertisingManifestsAsync(_includePreviews).Wait();
+            IEnumerable<(ManifestId manifestId, ManifestVersion existingVersion, ManifestVersion newVersion,
+                Dictionary<WorkloadDefinitionId, WorkloadDefinition> Workloads)> manifestsToUpdate =
+                _workloadManifestUpdater.CalculateManifestUpdates();
+
+            List<UpdateAvailableEntry> updateList = new();
+            foreach ((ManifestId _, ManifestVersion existingVersion, ManifestVersion newVersion,
+                Dictionary<WorkloadDefinitionId, WorkloadDefinition> workloads) in manifestsToUpdate)
             {
-                updateList.Add(new UpdateAvailableEntry("6.0.100", "6.0.101",
-                    _mockIosDescription,
-                    "microsoft-ios-sdk-full"));
+                foreach ((WorkloadDefinitionId workloadDefinitionId, WorkloadDefinition workloadDefinition) in
+                    workloads)
+                {
+                    if (installedWorkloads.Contains(new WorkloadId(workloadDefinitionId.ToString())))
```

Comment:
Won't we get "updates" advertised that are already installed with this logic? Should we compare manifest versions as well to make sure that this is really an update and not the version we already have installed?

Reply:
Talked offline. Updater will do a no op, but manifest will be updated. So still counts as an update.